src/cli/submit/SubmitJobHandler.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { JobUtils } from "../../api/JobUtils";
+import { FTPBaseHandler } from "../../FTPBase.Handler";
+import { IFTPHandlerParams } from "../../IFTPHandlerParams";
+
+const ONE_SECOND = 1000;
+const DEFAULT_INTERVAL = 5000;
+const DEFAULT_MAX_TRIES = 12;
+
+export abstract class SubmitJobHandler extends FTPBaseHandler {
+
+    public async submitJCL(jcl: string, params: IFTPHandlerParams): Promise<void> {
+        const jobid = await params.connection.submitJCL(jcl);
+        const jobDetails = await JobUtils.findJobByID(jobid, params.connection);
+        const message = params.response.console.log("Submitted job successfully, jobname(jobid): %s(%s)", jobDetails.jobname, jobDetails.jobid);
+        this.log.info(message);
+
+        if (params.arguments.wait) {
+            const message2 = params.response.console.log("Waiting for job completion.");
+            this.log.info(message2);
+
+            let interval = 0;
+            let maxTries = 0;
+            const input = RegExp(/^\d+,\d+$/);
```

Comment:
I like the idea for validating the input and not failing the command due to user error. 👍

Reply:
Thank you!